packages/embark-deployment/src/index.js

```diff
@@ -69,16 +69,15 @@ class DeployManager {
                 if (typeof result === 'function') {
                   callback = result;
                 }
-                if (contract.address && typeof contract.address === 'function') {
-                  contract.addressHandler = contract.address;
-                  delete contract.address;
+                if (contract.addressHandler) {
                   return self.plugins.runActionsForEvent('contract:address:handler', {contract}, (err, address) => {
                     if (err) {
                       errors.push(err);
                     } else {
                       contract.address = address;
                       contract.deployedAddress = address;
                       self.logger.info(__('{{contractName}} already deployed at {{address}}', {contractName: contract.className.bold.cyan, address: contract.address.bold.cyan}));
+                      self.events.emit("deploy:contract:deployed", contract);
```

Comment:
We emit this event here so that Dashboard can update its UI (this emission is what causes the error in the tracker).